subprojects/core/src/main/java/org/gradle/api/internal/file/copy/FileCopyDetailsInternal.java

```diff
@@ -18,9 +18,28 @@
 
 import org.gradle.api.file.FileCopyDetails;
 
+/**
+ * @implSpec Implementations representing a file copy of an entry in a compressed archive should return {@code true} from {@link #isCompressedArchiveEntry()}.
+ */
 public interface FileCopyDetailsInternal extends FileCopyDetails {
 
     boolean isDefaultDuplicatesStrategy();
 
     CopySpecResolver getSpecResolver();
+
+    /**
+     * Obtain a display name that represents what is being copied.
+     *
+     * @return display name
+     */
+    String getDisplayName();
+
+    /**
+     * Checks if this file copy details represents a file copy of an entry in a compressed archive.
+     *
+     * @return {@code true} if this file copy details represents a file copy of an entry in a compressed archive; {@code false} otherwise
+     */
+    default boolean isCompressedArchiveEntry() {
+        return false;
+    }
```

Comment:
❌ I don't think we should be exposing this.

Reply:
I'm getting rid of this, as there's no need after reworking the message, but would like to hear more - why is it a problem if the copy details internally knows whether it's copying from an archive or not?